tools/assets-automation/assets-maintenance-tool/Azure.Sdk.Tools.Assets.MaintenanceTool/Model/RepoConfiguration.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.Assets.MaintenanceTool.Model
+{
+    /// <summary>
+    /// Configuration class describing options available while targeting a repository for scanning.
+    /// </summary>
+    public class RepoConfiguration
+    {
+        public RepoConfiguration(string repo)
+        { 
+            Repo = repo;
+        }
+
+        /// <summary>
+        /// The full orgname/repo-id identifier to access a repo on github. EG: "azure/azure-sdk-for-net"
+        /// </summary>
+        public string Repo { get; set; }
```

Comment:
I do make use of these, but that's in a test, so I could definitely be convinced to update this. If I want to easily instantiate versions of this object in test, how would I do that while eliminating the setter?

Reply:
The basic pattern that I know of is:

Instead of:

```
var foo = new Foo()
foo.setBar(barVal)
foo.setQux(quxVal)
```

is to do:

```
var foo = new Foo(barVal, quxVal)
```

or even use [object initalizer](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer). There are some [nice lang keywords](https://github.com/dotnet/docs/issues/35014) supporting this.

In more advanced case, when you need to do some processing, like:

```
var foo = new Foo()
// do some stuff using Foo
var barVal = computeBarVal();
foo.setBar(barVal)
// do some stuff using Foo and barVal
var quxVal = computeQuxVal();
foo.setQux(quxVal)
```

then the answers is _it depends_ ;) The code could either be redesigned to never have to deal with partially initialized Foo e.g. by reordering control flow, or by making `barVal` and `quxVal` be computed lazily/just-in-time, and passing the functions that compute them as input to the ctor. But it really depends on the specific case.

Reply:
One more thing I wanted to add. Consider this:

```
var car = new CarChassis()

var engine = new Engine();
car.SetEngine(engine);

var seatBelts = new SeatBelts();

var Seats = new Seats();
seats.setSeatBealts(seatBelts);

car.setSeats(Seats);

var steeringWheel = new SteeringWheel();

var controlPanel = new ControlPanel();
controlPanel.setSteeringWheel(steeringWheel);

car.setControlPanel(controlPanel);
```

vs

```
var car = new CarChassis(
  engine: new Engine(),
  seats: new Seats(
    seatBelts: new SeatBelts()
  ),
  controlPanel: 
    new ControlPanel(
      steeringWheel: new SteeringWheel()
  )
)
```

I don't how about you, but for me the latter, declarative style reads so much better. And no setters required :)